src/python/zquantum/core/circuit/_circuit_test.py

```diff
@@ -96,13 +123,44 @@ def test_circuit_init(self):
 
         self.assertEqual(circuit_from_cirq, circuit_from_pyquil)
 
+    def test_circuit_evaluate(self):
+        # Given
+        theta_1 = sympy.Symbol("theta_1")
+        theta_2 = sympy.Symbol("theta_2")
+        value_1 = 0.5
+        value_2 = 0.6
+        symbols_map = [(theta_1, value_1), (theta_2, value_2)]
+        circuit = Circuit(
+            Program().inst(RX(2 * theta_1 + theta_2, 0), RY(theta_1, 0), RZ(theta_2, 0))
+        )
+        target_circuit = Circuit(
+            Program().inst(RX(2 * value_1 + value_2, 0), RY(value_1, 0), RZ(value_2, 0))
+        )
+
+        # When
+        evaluated_circuit = circuit.evaluate(symbols_map)
+
+        # Then
+        self.assertEqual(evaluated_circuit, target_circuit)
+
```

Comment:
There should also be test cases for:
- calling `evaluate` on a circuit without any symbolic params
- calling `evaluate` with some but not all of the symbolic params specified
- calling `evaluate` with an empty symbols map or a map with symbols not in the circuit -> what is the intended behavior here
- calling `evaluate` with more symbols in the map than in the circuit

Reply:
Adressed in [a2bda64](https://github.com/zapatacomputing/z-quantum-core/pull/39/commits/a2bda6408bda9011406fc198d2b194ee04b7d6f8)